### DISCUSSION TITLE
Several osu!catch editor bugs and issues making it unfun and impractical to use

### DISCUSSION THREAD

**Discussion by Akariow** [ppy/osu#27771] [Q&A]
### Type

Other

### Bug description

I wanted to get accustomed to mapping catch in lazer, so I tried using it for a few weeks and want to report my bad experiences with it here since no one else even dares to use it. Fundamentally its a good editor but more niche things and other issues make it very displeasing to continue using.

## General:
Bookmarks: I always found bookmarks (blue bars) very useful to structure my map with to the song and quickly move between significant parts. Having them again would be a useful addition.

BPM Setting: Allow for 2 or 3 decimal places of accuracy as a single decimal place limits precision and can still results in multiple milliseconds of error if the BPM is off by 0.05 making.

Hitsounding: Its currently not possible to add a hitsounds to specifics a Fruit of a JuiceStream. And I would assume the same is the case for sliders in Std.
![image](https://github.com/ppy/osu/assets/109684187/9891d586-2b8b-4647-9fcf-b04d0ae4f460)
![image](https://github.com/ppy/osu/assets/109684187/6111011e-038a-4089-891b-0db9c851b10a)

## Catch specific
Visual Slider Tick/Velocity bug: Has been reported before and forces the user to reload the editor every time they change a slider in order to make it visually correct. This really messes with the flow of mapping.

Very Loud skipping bug: Sometimes when moving along the timeline with arrow keys a lot of objects will be loaded and *catched* at once and make an incredibly loud noise, this gets very annoying via audio and visuals when having to rewind often or skipping trough the map a lot.
I have not yet replicated it with arrow key timeline navigation on video, it just kinda starts happening after mapping for a while I will try and get a recording of it if, it happens again. With skipping trough the map its replicable.
https://github.com/ppy/osu/assets/109684187/338201b4-6fd4-4c60-a335-4ab34bd1018d

Exporting Maps: An important feature as maps can not yet be uploaded from lazer, *unclean* JuiceStreams with many nodes that are not close to being snapped or JuiceStreams with a high slider velocity can break when exporting.
![image](https://github.com/ppy/osu/assets/109684187/57fbac20-5aef-4a12-9219-fd90b17f8237)
https://github.com/ppy/osu/assets/109684187/001a41c4-9992-4029-86b8-cb77586ecef6

Inconsistent Fruit Droplet generation: For 1/2 sliders for 200 BPM, and possibly more scenarios, this affects gameplay and is inconsistent with stable! Id assume it has something to do with a float check and some values are slightly over or under some limit for 220 bpm specifically.
https://github.com/ppy/osu/assets/109684187/8f75826b-67b1-4dec-8af5-138056028d7f

I hope this is written comprehensively enough, the lazer editor has great potential but its still being held back by many of such issues, many mappers ive talked to do not want to use it for the listed reasons, I hope soon this can be improved on and start bringing more people over to lazer.

### Version

2024.312.1

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14842416/compressed-logs.zip)

**Comment by 64ArthurAraujo** [comment]
> Bookmarks: I always found bookmarks (blue bars) very useful to structure my map with to the song and quickly move between significant parts. Having them again would be a useful addition.

I also feel the need of bookmarks in the editor, importing a stable map with bookmarks and importing to stable from lazer wipes them. I would be up to implementing them, also i think it would be cool to have bookmark labels as suggested by some other previous discussions.

**Comment by bdach** [comment]
Received this via other routes so just going to dump this here for later tracking: https://docs.google.com/document/d/18PjArFl5NTNTyXA1zpDGoK4Yl4CEh2pVjxiJetFEXA0/edit
